### ISSUE TITLE
[4.0] Alert Warning: insufficient colour contrast ratio

### ISSUE THREAD

**Issue by chmst** [joomla/joomla-cms#30375]
### Steps to reproduce the issue
Try to login with a wrong password. 
You will get a warning. 
Check the site with AXE of WAVE or your favourite a11y check tool. 

### Expected result
No contrast error

### Actual result
The warning / text combination has a contrast ratio of about 1.7.

**Comment by brianteeman** [comment]
Are you referring to logging in to the admin or to the site?

**Comment by chmst** [comment]
It is for both. Defined in the webcomponent for alerts.

**Comment by Quy** [comment]
Duplicate #29174?

**Comment by chmst** [comment]
#29174 should be resolved now. If I understand right, it addresses the alerts in bootstrap and meanwhile they have good contrast ratio. #27369

![screen shot 2020-08-15 at 14 17 58](https://issues.joomla.org/uploads/1/0dcfcb88a57664cea181170720086b1d.png)

My issue is caused by messages of type warning from the system, this is undpendent from client. 

![screen shot 2020-08-15 at 14 17 58](https://issues.joomla.org/uploads/1/106a24aa09443f4e5c5f778638b8eacc.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30375">issues.joomla.org/tracker/joomla-cms/30375</a>.</sub>

**Comment by infograf768** [comment]
AFAIK it is defined in atum and cassiopea since https://github.com/joomla/joomla-cms/pull/30294 by
Atum `/administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-alert.scss`
and similar for Cassiopea
```
&[type="warning"] {
    --alert-accent-color: var(--warning);
  }
```
in Atum
`var(--warning)` equals `$yellow` in variables.
and `$yellow:                           #ffb514; //used in bootstrap`

and in Cassiopea by 
`var(--warning) equals $yellow in variables.`
and `$yellow:                             #f0ad4e;`

Therefore a small difference for the `$yellow` which could be normalised and modified if needed.

**Comment by infograf768** [comment]
We could use as color `#cd7b07` which would give a correct contrast
<img width="933" alt="Screen Shot 2020-08-15 at 18 58 40" src="https://user-images.githubusercontent.com/869724/90317396-7783c900-df29-11ea-97de-bf2b5f778de8.png">

What do you think?

**Comment by chmst** [comment]
It is an issue in frontend and backend. Your proposal looks good. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30375">issues.joomla.org/tracker/joomla-cms/30375</a>.</sub>

**Comment by infograf768** [comment]
Now remains to decide if we use a new color directly for `$warning` or modify the `$yellow` in variables for each template.
I am not a specialist of a11y and css. I leave this to whoever is.

**Comment by infograf768** [comment]
Hmm
It looks like Wave does not agree with https://webaim.org/resources/contrastchecker/ and firefox own contrast check

I modified here the text color for the Warning title to get black on the existing yellow instead of modifying the `$yellow`
I get

<img width="1186" alt="Screen Shot 2020-08-15 at 20 01 13" src="https://user-images.githubusercontent.com/869724/90318645-2b895200-df32-11ea-9f64-df24395790cb.png">

and contrastchecker agrees

![Screen Shot 2020-08-15 at 20 02 31](https://user-images.githubusercontent.com/869724/90318662-4cea3e00-df32-11ea-95d7-3c775a2902ef.png)

Who should we believe?

Modifcations in the scss to get black text for warning are easy. Just create a new variable `var(--alert-heading-text)` and adapt to warning type
```
  .alert-heading {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    padding: .8rem;
    color: var(--alert-heading-text);
    //color: rgba(255, 255, 255, .95);
    background: var(--alert-accent-color, transparent);
  }

  .alert-link {
    color: var(--success, inherit);
  }

  &[type="success"] {
    --alert-accent-color: var(--success);
    --alert-heading-text: rgba(255, 255, 255, .95);
  }

  &[type="info"] {
    --alert-accent-color: var(--info);
    --alert-heading-text: rgba(255, 255, 255, .95);
  }

  &[type="warning"] {
    --alert-accent-color: var(--warning);
    --alert-heading-text: #000;
  }

  &[type="danger"] {
    --alert-accent-color: var(--danger);
    --alert-heading-text: rgba(255, 255, 255, .95);
  }
```

**Comment by infograf768** [comment]
I rather believe contrastchecker and Firefox check.
Will do PR to change text to black for the Warning.

It is what Brian did anyway here: https://github.com/joomla/joomla-cms/pull/27369

**Comment by brianteeman** [comment]
Thanks for reverting to my original code

Odd that  Wave does not agree with https://webaim.org/resources/contrastchecker/  because they are both made by webaim

**Comment by chmst** [comment]
It is one of these things which drive me crazy

> Odd that Wave does not agree with https://webaim.org/resources/contrastchecker/ because they are both made by webaim

**Comment by infograf768** [comment]
> Thanks for reverting to my original code

It is not really a revert as the code to get the present results is quite different.
Just something forgotten for the warning case.

**Comment by infograf768** [comment]
Please test https://github.com/joomla/joomla-cms/pull/30382
Closing as we have a patch
